mldsa/rounding.h

```diff
@@ -10,7 +10,31 @@
 #include "params.h"
 
 #define power2round MLD_NAMESPACE(power2round)
-int32_t power2round(int32_t *a0, int32_t a);
+/*************************************************
+ * Name:        power2round
+ *
+ * Description: For finite field element a, compute a0, a1 such that
+ *              a mod^+ MLDSA_Q = a1*2^MLDSA_D + a0 with -2^{MLDSA_D-1} < a0 <=
+ *              2^{MLDSA_D-1}. Assumes a to be standard representative.
+ *
+ * Arguments:   - int32_t a: input element
+ *              - int32_t *a0: pointer to output element a0
+ *              - int32_t *a1: pointer to output element a1
+ *
+ * Reference: a1 is passed as a return value instead
+ **************************************************/
+void power2round(int32_t *a0, int32_t *a1, int32_t a)
+__contract__(
+  requires(memory_no_alias(a0, sizeof(int32_t)))
+  requires(memory_no_alias(a1, sizeof(int32_t)))
+  requires(a >= 0 && a < MLDSA_Q)
+  assigns(memory_slice(a0, sizeof(int32_t)))
+  assigns(memory_slice(a1, sizeof(int32_t)))
+  ensures(*a0 > -(1<<MLDSA_D) && *a0 <= (1<<MLDSA_D))
```

Comment:
General comment: Does it make sense to introduction a macro-abbreviation for `1<<MLDSA_D`?

Reply:
It appears on 3 lines in rounding.h, so I'd say Yes...

Reply:
okay - I added `MLD_2_POW_D`